R/htmlwidgets.R

```diff
@@ -182,13 +182,15 @@ toHTML <- function(x, standalone = FALSE, knitrOptions = NULL) {
     style = css(
       width = validateCssUnit(sizeInfo$width),
       height = validateCssUnit(sizeInfo$height),
+    ),
+    class = paste(
+      name, "html-widget",
       # bindFillRole() puts `overflow:auto` to items by default,
       # which is a sensible generic default, but in the context of
       # widgets, we can pretty easily run into non-pixel-perfect situations
       # (e.g., the widget JS uses something like offsetHeight to resize itself)
-      "--html-fill-item-overflow" = if (sizeInfo$fill) "hidden"
+      if (sizeInfo$fill) "html-fill-item-overflow-hidden"
```

Comment:
I think it might be better to not use `paste()` with conditionals inside; it results in extra spaces:

```r
> paste("1", NULL, "2")
[1] "1  2"
```

It's not that big of a deal but it's also very easy to use the approach where you have multiple `class` arguments.

Reply:
> it's also very easy to use the approach where you have multiple class arguments.

I don't think we can actually since this function can be provided by widget authors and have both `class` and `...` in the formal arguments (see the bottom of http://www.htmlwidgets.org/develop_advanced.html)